python/pyspark/sql/avro/functions.py

```diff
@@ -80,6 +81,25 @@ def from_avro(
     from py4j.java_gateway import JVMView
     from pyspark.sql.classic.column import _to_java_column
 
+    if not isinstance(data, (Column, str)):
+        raise PySparkTypeError(
+            error_class="INVALID_TYPE",
+            message_parameters={
+                "arg_name": "data",
+                "arg_type": ":class:`~pyspark.sql.Column` or str",
```

Comment:
`:class:...` is sphinx directive so we won't need to expose it in error message.

```suggestion
                "pyspark.sql.Column or str",
```